### DISCUSSION TITLE
Documentation for PDFViewer

### DISCUSSION THREAD

**Discussion by mark-norgate-bzy** [mozilla/pdf.js#18344] [Q&A]
I am trying to write an Angular component using pdf.js. I am getting so far, but then I keep discovering new corners and features of this library.

Trouble is, I cannot find any definitive documentation _anywhere_.

Where can I find this? There are dozens of web components in [this folder](https://github.com/mozilla/pdf.js/tree/master/web), for example, but none of them are documented. Am I really supposed to read the source code to work out what they do??

**Comment by BowFarmer** [comment]
Welcome to the club. I doubt the number of programmers working on the project have the time to write detailed documentation and to keep it maintained. That is a huge task. However, I find that I can easily enter a thousand lines of js into ChatGPT and ask it to summarize what that js does. It does a pretty good job.
Ask ChatGPT: "I am trying to write an Angular component using Mozilla's pdf.js but am running into problems. Can you help?"
You might be surprised at the reply.
